chain/chain/src/chain.rs

```diff
@@ -62,8 +50,20 @@ use near_primitives::views::{
 };
 use near_store::{ColState, ColStateHeaders, ColStateParts, ShardTries, StoreUpdate};
 
-#[cfg(feature = "delay_detector")]
-use delay_detector::DelayDetector;
+use crate::lightclient::get_epoch_block_producers_view;
+use crate::migrations::check_if_block_is_valid_for_migration;
+use crate::missing_chunks::{BlockLike, MissingChunksPool};
+use crate::store::{ChainStore, ChainStoreAccess, ChainStoreUpdate, GCMode};
+use crate::types::{
+    AcceptedBlock, ApplyTransactionResult, Block, BlockEconomicsConfig, BlockHeader,
+    BlockHeaderInfo, BlockStatus, ChainGenesis, Provenance, RuntimeAdapter,
+};
+use crate::validate::{
+    validate_challenge, validate_chunk_proofs, validate_chunk_with_chunk_extra,
+    validate_transactions_order,
+};
+use crate::{byzantine_assert, create_light_client_block_view, Doomslug};
+use crate::{metrics, DoomslugThresholdMode};
```

Comment:
@matklad do you have suggestions on how we can avoid re-arranging imports indefinitely :)

Reply:
I think two strategies work for big projects for dealing with "code normalization", including imports:

* just don't care about imports, let everyone arrange & rearrage them however they like
* make sure that all existing code is using consistent imports, so that the path of the least resistance for the new code is to be consistent. 

To make the second case work best, you need 

* simple opinionated rules which doesn't leave much place for judgement
* de-facto usage of those rule by the majority of the project
* some document which spells the rules out, to like during code reviews

rust-analyzer uses the second approach, it works OK. Here are our rules: https://github.com/rust-analyzer/rust-analyzer/blob/ad0648dc95d2d1e9d35cd825c2661fe002c02fc9/docs/dev/style.md#order-of-imports

At some point, this might be enforceable via rustfmt (https://github.com/rust-lang/rustfmt/issues/2979), but it doesnt' look like it'll be done soon. 

Practically, for nearcore, I suggest:

* Creating `style.md` document, and writing down *some* strategy we use (even if it is "don't care, don't touch existing imports")
* Maybe looking at the existing code and writing down a strategy we aspire to, and start passively normalizing the code to that. I think that might work out in the long-run.